rm-datasource/src/main/java/io/seata/rm/datasource/exec/AbstractDMLBaseExecutor.java

```diff
@@ -96,10 +96,10 @@ protected T executeAutoCommitFalse(Object[] args) throws Exception {
         if (!JdbcConstants.MYSQL.equalsIgnoreCase(getDbType()) && isMultiPk()) {
             throw new NotSupportYetException("multi pk only support mysql!");
         }
-        TableRecords beforeImage = beforeImage();
         T result = statementCallback.execute(statementProxy.getTargetStatement(), args);
         int updateCount = statementProxy.getUpdateCount();
         if (updateCount > 0) {
+            TableRecords beforeImage = beforeImage();
```

Comment:
the values of the before and after images are the same ?

Reply:
+1